src/extra/components/adq.py

```diff
@@ -0,0 +1,1294 @@
+        mode, where individual charged particles impact a detection
+        surface and leave a fast signal on the digitizer trace. Fast
+        timing discriminators allow to robustly determine the position
+        of such signals.
+
+        By default, it uses the
+        [dynamic leading discriminator](extra.signal.dled) from the
+        [extra.signal](extra.signal) package, but other from this
+        package or entirely custom functions may be used as well. The
+        required signatures must include three keyword arguments
+        `signal`, `edges` and `amplitudes` corresponding to those from
+        [extra.signal.dled](extra.signal.dled). The default edge finding
+        method requires the `threshold` parameter to be passed as
+        keyword argument.
+
+        The processing is parallelized via `pasha`.
+
+        Args:
+            data (ArrayLike): Input data to find edges on.
+            edge_func (Callable, optional): Edge finding method to run
+                on each train trace, extra.signal.dled by default.
+            max_edges (int, optional): Maximal number of edges per
+                train, 1/5000 of trace length by default.
+            with_amplitudes (bool, optional): Full pulse height
+                corresponding to the determined edge positions.
+            sample_dim ({sample, time}, optional): Coordinates for
+                sample dimension if a labelled result is returned,
+                'sample' by default.
+            shape ({edges, input}, optional): Whether to keep a single
+                edge dimension (`edges`, default) or preserve the input
+                shape (`input`). The latter option returns labelled
+                xarray objects, while the former returns pandas
+                series or dataframe.
+            squeeze_edges (bool, optional): Whether to minimize the edge
+                axis lengh for xarray objects, True by default and
+                ignored when pandas objects are returned.
+            pasha_ctx (pasha.MapContext, optional): Pasha context to run
+                parallelized operations, pasha.ProcessContext by default.
+
+            Any further keyword arguments are passed to the edge finding
+            method.
+
+        Returns:
+            result (pd.Series): Edge positions, only if edge shaped and
+                without amplitudes.
+
+            result (pd.DataFrame): Edge positions and pulse heights,
+                only if edge shaped and with amplitudes.
+
+            result (xarray.DataArray): Edge positions, only if train or
+                pulse shaped and without amplitudes.
+
+            result (xarray.Dataset): Edge positions and pulse heights,
+                only if train or pulse shaped and with amplitudes.
+        """
+
+        # When given or returning xarray, include digitizer parameters
+        # as attributes
+
+        if _isinstance_no_import(data, 'xarray', 'DataArray'):
```

Comment:
This is TBD, but seemed less relevant for now.

Reply:
If TBD, maybe throw a NotImplementedError or something?

Reply:
Passing an `xarray` still works and computes the correct result, but will replace the coordinates by `dim_N` rather than the original. I would not think that warrants breaking the method completely in this case.